### PR TITLE
Feature/openlayers amplify

### PR DESCRIPTION
**Pull request by dayjournal**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

This example demonstrates how to use OpenLayers with Amazon Location Service and AWS Amplify.
https://dev.to/aws-heroes/building-a-map-application-with-amazon-location-service-openlayers-and-aws-amplify-2f54

### CODE REVIEW

**Comment by dreamorosi** [comment]
Hi @dayjournal, thank you for opening this PR.

Please give me a few days to review the PR and provide comments.

I'll be in touch here once I have reviewed, but you can expect a reply by early next week at the latest.

**Comment by dayjournal** [comment]
Hi, @dreamorosi , OK!
Please let us know if you have any questions.

File: openlayers-amplify/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- npm v8.x
+- Amplify CLI v9.2.x
+
+## Getting started
+
+1. Install the Amplify CLI: `npm install -g @aws-amplify/cli`
+2. Install project dependencies: `npm install`
+3. Create a new Amplify project environment: `amplify init`
+4. Create Auth and Geo resources: `amplify push`
+5. After deployment is complete, change the extension of `/src/aws-exports.js` to `/src/aws-exports.ts`
```
Comment (dreamorosi) [inline openlayers-amplify/README.md:21]: Let's please remove this step.

Instead, let's add this line to the project's `tsconfig.json`:
```
"allowJs": true,
```

This way the IDE won't complain about the missing types of the `aws-exports` file.

`aws-exports.js` is a file that gets automatically generated and gets overwritten every time the user runs `amplify push`, so better not to rename it.

File: openlayers-amplify/vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      './runtimeConfig': './runtimeConfig.browser',
+    },
+  },
+  define: {
+    'window.global': {}
+  },
+  build: {
+    target: 'esnext'
+  },
+})
```
Comment (dreamorosi) [inline openlayers-amplify/vite.config.ts:15]: Please include the `server` configs from [this other sample](https://github.com/aws-samples/amazon-location-samples/blob/main/maplibre-gl-js-vue-amplify/vite.config.js#L14-L17), it'll help with customers who are running this on browser-based IDEs like Cloud9.

**Review by dreamorosi — CHANGES_REQUESTED** [review]
I have reviewed the sample and it looks already in a great shape.

I have left only two minor comments related to configurations, once these are addressed I'll be happy to approve & merge this sample.

File: openlayers-amplify/README.md (hunk 1 above)
Comment (dayjournal) [reply openlayers-amplify/README.md]: Fixed.

File: openlayers-amplify/vite.config.ts (hunk 2 above)
Comment (dayjournal) [reply openlayers-amplify/vite.config.ts:19]: Fixed.

**Comment by dayjournal** [comment]
Hi, @dreamorosi. I have modified my PR.

**Review by dreamorosi — APPROVED** [review]
Thank you @dayjournal for contributing this sample and addressing the review comments.

I'll be merging this!

**Comment by dayjournal** [comment]
@dreamorosi Thank you!!

**Pull request merged**